bigtop-deploy/puppet/modules/ranger/manifests/init.pp

```diff
@@ -17,57 +17,77 @@
+
+  class admin($admin_password) {
     # Before Facter 3.14.17, Rocky Linux 8 is detected as 'RedHat'.
     # https://puppet.com/docs/pe/2019.8/osp/release_notes_facter.html#enhancements-3-14-17
-    if (($operatingsystem == 'RedHat' or $operatingsystem == 'Rocky') and 0 <= versioncmp($operatingsystemmajrelease, '8')) {
-      # For some reason, 'python3' doesn't seem to work on Rocky Linux 8.
-      $python = 'python36'
-    } else {
-      $python = 'python3'
+    case $operatingsystem {
+      /(?i:(ubuntu|debian))/: {
+            $postgres_packages = ['postgresql']
+            $python = 'python3'
+            $java_home_env = 'JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-amd64'
+        }
+        default: {
+            $postgres_packages = ['postgresql-jdbc', 'postgresql-server']
+            $python = 'python36'
+            $java_home_env = 'JAVA_HOME=/usr/lib/jvm/java-1.8.0'
+        }
     }
 
-    package { ['postgresql-jdbc', 'postgresql-server', $python]:
-      ensure => latest,
+    package { ['ranger-admin', $python]:
+          ensure => latest,
     }
 
-    exec { 'initdb':
-      command => '/usr/bin/pg_ctl initdb -D /var/lib/pgsql/data',
-      user    => 'postgres',
-      require => Package['postgresql-jdbc', 'postgresql-server', $python],
-    }
+    if ($operatingsystem =~ /^(?i:(ubuntu|debian))$/) {
+      notice("Matched ubuntu or debian")
+      service { 'postgresql':
+        ensure  => running,
+        require => Package[$python, 'ranger-admin'],
+      }
+
+      file { '/usr/share/java/postgresql-jdbc.jar':
+        ensure => link,
+        target => '/usr/share/java/postgresql.jar',
+        require => Package[$python, 'ranger-admin'],
+      }
+    } else {
+      notice("Did not match ubuntu or debian")
+      exec { 'initdb':
+        command => '/usr/bin/pg_ctl initdb -D /var/lib/pgsql/data',
+        user    => 'postgres',
+        require => Package[$python, 'ranger-admin'],
+      }
 
-    service { 'postgresql':
-      ensure  => running,
-      require => Exec['initdb'],
+      service { 'postgresql':
+        ensure  => running,
+        require => Exec['initdb'],
+      }
     }
-  }
 
-  class admin($admin_password) {
-    package { 'ranger-admin':
-      ensure  => latest,
-      require => Class['ranger::prerequisites'],
+    exec { 'change_postgres_password':
+      command => "/bin/sudo -u postgres /usr/bin/psql -c \"ALTER USER postgres WITH PASSWORD 'admin';\"",
+      require => Service['postgresql'],
     }
 
+    notice("Before defining file resource")
     file { '/usr/lib/ranger-admin/install.properties':
       content => template('ranger/ranger-admin/install.properties'),
-      require => Package['ranger-admin'],
+      require => [Package['ranger-admin'], Exec['change_postgres_password']]
     }
 
     exec { '/usr/lib/ranger-admin/setup.sh':
       cwd         => '/usr/lib/ranger-admin',
-      environment => 'JAVA_HOME=/usr/lib/jvm/java-1.8.0',
+      environment => $java_home_env,
```

Comment:
I understand the original manifest used hard-coded JAVA_HOME, but basically we should use bigtop-detect-javahome provided by bigtop-utils to locate JAVA_HOME whenever we can, so that we don't need to have duplicated logic. For example, running setup.sh and set_globals.sh with JAVA_HOME here can be rewrited as follows. (That being said, the following implementation is ugly... If you find other solution, feel free to adopt it)

```
    exec { 'setup.sh':
      command => '/bin/bash -c "source /usr/lib/bigtop-utils/bigtop-detect-javahome && /usr/lib/ranger-admin/setup.sh"',
      cwd     => '/usr/lib/ranger-admin',
      require => File['/usr/lib/ranger-admin/install.properties'],
    }

    exec { 'set_globals.sh':
      command => '/bin/bash -c "source /usr/lib/bigtop-utils/bigtop-detect-javahome && /usr/lib/ranger-admin/set_globals.sh"',
      cwd     => '/usr/lib/ranger-admin',
      require => Exec['setup.sh'],
    }

    exec { 'systemctl daemon-reload':
      path    => ["/bin", "/usr/bin"],
      require => Exec['set_globals.sh'],
    }
```